ksmt-core/src/main/kotlin/io/ksmt/KContext.kt

```diff
@@ -1334,9 +1650,28 @@ open class KContext(
     // integer
     private val intModCache = mkAstInterner<KModIntExpr>()
 
+    /**
+     * Create an Int mod expression.
+     *
+     * The result value is a number `r` such that
+     * [lhs] = `r` + ([rhs] * (div [lhs] [rhs]))
+     * where `div` is an Int division that works according to the [mkArithDiv] rules.
+     * The result value is always positive or zero.
+     *
+     * For example:
+     *   `47 mod 13 = 8`
+     *   `47 mod -13 = 8`
```

Comment:
![image](https://github.com/UnitTestBot/ksmt/assets/31047452/f057cbf9-1838-4661-a844-4e5dfa9ae81b)

Reply:
There is an issue in KSMT simplifier. 
```
    @Test
    fun testIntMod(): Unit = with(KContext()) {
        KZ3Solver(this).use { solver ->
            val a = mkIntNum(47)
            val b = mkIntNum(-13)

            val r = mkConst("r", intSort)
            solver.assert(r eq mkIntModNoSimplify(a, b))
            solver.check()
            val z3Res = solver.model().eval(r, isComplete = false)

            val ksmtRes = mkIntMod(a, b)

            println(z3Res) // 8
            println(ksmtRes) // -5
        }
    }
```